frontend/src/metabase/containers/AddToDashSelectDashModal/hooks.ts

```diff
@@ -0,0 +1,47 @@
+import { useAsync } from "react-use";
+import type { CollectionId, Dashboard } from "metabase-types/api";
+import { ActivityApi } from "metabase/services";
+import { ROOT_COLLECTION } from "metabase/entities/collections";
+import { coerceCollectionId } from "metabase/collections/utils";
+
+export const useMostRecentlyViewedDashboard = () => {
+  const {
+    loading: isLoading,
+    error,
+    value: data,
+  } = useAsync(async () => {
+    const dashboard: Dashboard | undefined =
+      await ActivityApi.most_recently_viewed_dashboard();
+
+    return dashboard;
+  });
+
+  return { data, isLoading, error };
+};
+
+interface UseCollectionIdProps {
+  isQuestionInPersonalCollection: boolean;
+  mostRecentlyViewedDashboard: Dashboard | undefined;
+}
+
+export const useCollectionId = ({
```

Comment:
new hook

Reply:
nit: improve naming, e.g. add some more info about what the "collection" is here

Reply:
How about `useInitialOpenCollectionId`. We have the same variable name elsewhere which does exactly the same thing.

https://github.com/metabase/metabase/blob/13424d36f2fef49c1884a7f917a1e1febb337be0/frontend/src/metabase/collections/containers/FormCollectionPicker/FormCollectionPicker.tsx#L34

It conveys that this is the value for the open collection when first rendering the picker.

Reply:
Fixed in 7f631ecb94be97214aeb1bc6c2233760a1e93a3e